timeseries/src/autogluon/timeseries/models/ensemble/abstract.py

```diff
@@ -0,0 +1,136 @@
+
+logger = logging.getLogger(__name__)
+
+
+class AbstractTimeSeriesEnsembleModel(TimeSeriesModelBase, ABC):
+    """Abstract class for time series ensemble models."""
+
+    @property
+    def model_names(self) -> List[str]:
+        """Names of base models included in the ensemble."""
+        raise NotImplementedError
+
+    @final
+    def fit(
+        self,
+        predictions_per_window: Dict[str, List[TimeSeriesDataFrame]],
+        data_per_window: List[TimeSeriesDataFrame],
+        model_scores: Optional[Dict[str, float]] = None,
+        time_limit: Optional[float] = None,
+    ):
+        """Fit ensemble model given predictions of candidate base models and the true data.
+
+        Parameters
+        ----------
+        predictions_per_window : Dict[str, List[TimeSeriesDataFrame]]
+            Dictionary that maps the names of component models to their respective predictions for each validation
+            window.
+        data_per_window : List[TimeSeriesDataFrame]
+            Observed ground truth data used to train the ensemble for each validation window. Each entry in the list
+            includes both the forecast horizon (for which the predictions are given in ``predictions``), as well as the
+            "history".
+        model_scores : Optional[Dict[str, float]]
+            Scores (higher is better) for the models that will constitute the ensemble.
+        time_limit : Optional[int]
+            Maximum allowed time for training in seconds.
+        """
+        if time_limit is not None and time_limit <= 0:
+            logger.warning(
+                f"\tWarning: Model has no time left to train, skipping model... (Time Left = {round(time_limit, 1)}s)"
+            )
+            raise TimeLimitExceeded
+        if isinstance(data_per_window, TimeSeriesDataFrame):
+            raise ValueError("When fitting ensemble, `data` should contain ground truth for each validation window")
+        num_val_windows = len(data_per_window)
+        for model, preds in predictions_per_window.items():
+            if len(preds) != num_val_windows:
+                raise ValueError(f"For model {model} predictions are unavailable for some validation windows")
+        self._fit(
+            predictions_per_window=predictions_per_window,
+            data_per_window=data_per_window,
+            model_scores=model_scores,
+            time_limit=time_limit,
+        )
+        return self
+
+    def _fit(
+        self,
+        predictions_per_window: Dict[str, List[TimeSeriesDataFrame]],
+        data_per_window: List[TimeSeriesDataFrame],
+        model_scores: Optional[Dict[str, float]] = None,
+        time_limit: Optional[float] = None,
+    ):
+        """Private method for `fit`. See `fit` for documentation of arguments. Apart from the model
+        training logic, `fit` additionally implements other logic such as keeping track of the time limit.
+        """
+        raise NotImplementedError
```

Comment:
Nit: Should this be an `@abstractmethod` instead of raising `NotImplementedError`?